docs/basics/transactions.md

```diff
@@ -326,6 +326,9 @@ following aspects of a transaction to be configured:
     cache access during read operations. Doesn't have any effect if database 
     level cache was disabled via config `cache.db-cache`.
 
+-   `lazyLoadRelations()` - Set lazy-load for all properties and edges of the vertex: ids and values are deserialized upon demand.
+    When enabled, it can have a boost on large-scale read operations, if only certain type of relations are being read from the vertex.
```

Comment:
nit: type -> types

Reply:
I realize I don't actually know what you mean. An example would be helpful.

Reply:
I also realize that I have quickly forgot a lot about JanusGraph internals :( But now I have a fresh view as an ordinary user :P

Reply:
The lazy load helps when you are fetching vertex from data storage and loading all properties and edges, but you only need to access 2-3 properties out of this collection.
The current relation iterator is implemented in a way that it deserializes every single relation to find the one requested by the query, making the traversal inefficient. Instead, you can still filter by relation type (edge label or property key) without deserializing the actual value of the relation if you don't need it.
https://github.com/JanusGraph/janusgraph/blob/ab3a80123844976382dc43acda6d8aaf0ff29183/janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/RelationConstructor.java#L45

Reply:
After this PR, I wanted to raise a different one, which will expose a new API to take full advantage of lazy loading.

https://github.com/JanusGraph/janusgraph/blob/ab3a80123844976382dc43acda6d8aaf0ff29183/janusgraph-core/src/main/java/org/janusgraph/core/JanusGraphMultiVertexQuery.java#L123

```
 /**
     * Returns an iterable overall incident properties that match this query for each vertex.
     * Properties are loaded once on property map value access.
     *
     * @return Iterable over all incident properties that match this query for each vertex
     */
Map<JanusGraphVertex, Map<String, LazyVertexProperty>> propertiesMap();
```